core/src/dht/dht_store.rs

```diff
@@ -61,6 +66,51 @@ impl DhtStore {
         )
     }
 
+    /// Get all headers for an entry by first looking in the DHT meta store
+    /// for header addresses, then resolving them with the DHT CAS
+    pub fn get_headers(&self, entry_address: Address) -> Result<Vec<ChainHeader>, HolochainError> {
+        self.meta_storage()
+            .read()
+            .unwrap()
+            // fetch all EAV references to chain headers for this entry
+            .fetch_eavi(
+                Some(entry_address),
+                Some(ENTRY_HEADER_ATTRIBUTE.to_string()),
+                None,
+                Default::default(),
+            )?
+            .into_iter()
+            // get the header addresses
+            .map(|eavi| eavi.value())
+            // fetch the header content from CAS
+            .map(|a| self.content_storage().read().unwrap().fetch(&a))
```

Comment:
any way to not unwrap here?

Reply:
@thedavidmeister Didn't we agree that it's OK to unwrap when locking a mutex or rwlock? All the unwraps are of that variety.

Reply:
I'd like to defer on all the unwraps until we get our error handling story straight (https://github.com/holochain/holochain-rust/issues/949)